### DISCUSSION TITLE
When adding a system environment variable with vercel env, automatically pick the value if the name matches a system environment variable name

### DISCUSSION THREAD

**Discussion by henderea** [vercel/vercel#5669] [Ideas]
When adding a system environment variable at command line, setting the name of the environment variable to the name of a system environment variable value option should automatically select the referenced system environment variable.

For example, if I run `vercel env add system VERCEL_URL production`, I shouldn't have to answer any prompts, because I'm clearly referring to the `VERCEL_URL` system environment variable.  If you don't want to force the assumption, you could ask a yes/no question "Use the system value VERCEL_URL?" and have a `--yes` or `--confirm` flag for the command so I can skip the prompt.

Alternatively, allow me to specify the system environment variable as part of the command, such as `vercel env add system VERCEL_URL production --value=VERCEL_URL` or `vercel env add system VERCEL_URL=VERCEL_URL production` or some other style.

Lately, vercel has been getting more and more inconvenient.  I used to be able to specify everything in a JSON file and deploy with a single command.  If you're not going to bring that back, at least make the CLI more compatible with scripting so I can automate some of my default setup.

**Comment by styfle** [comment]
Hi @henderea 

We made some changes to Environment Variables.

Let me know if this solves your use case https://vercel.com/changelog/environments-variables-per-git-branch

**Reply by henderea** [reply]
No, that seems completely unrelated.  I don't even _use_ the git integration.  Also, I was talking about making it easier to add system environment variables at command-line, since I add the same one to every project and would like to be able to script it.

**Reply by styfle** [reply]
You shouldn't need to manually add system env vars anymore.

We have a checkbox to automatically expose those here: https://vercel.com/docs/environment-variables#system-environment-variables

**Reply by henderea** [reply]
Well, unless I can turn that on for all new projects automatically, or turn it on from the command line, it doesn't cover my needs.  I'm looking to be able to automate this for new projects with a script.

**Reply by styfle** [reply]
> turn that on for all new projects automatically

It is indeed enabled for all new projects! 🙂
